platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -199,4 +130,82 @@ private void processExistingConfigurationFiles() {
     Stream<Path> getExistingFiles() throws IOException {
         return Files.walk(configurationDirectoryPath);
     }
+
+    private class ListenerImpl implements FileAlterationListener {
+        @Override
+        public void onStart(FileAlterationObserver fileAlterationObserver) {
+        }
+
+        @Override
+        public void onDirectoryCreate(File file) {
+        }
+
+        @Override
+        public void onDirectoryChange(File file) {
+        }
+
+        @Override
+        public void onDirectoryDelete(File file) {
+        }
+
+        @Override
+        public void onFileCreate(File file) {
+            try {
+                String filename = file.getName();
+                LOGGER.debug("Watcher has been notified. Handling event for [{}].", filename);
+
+                if (!filename.endsWith(fileExtension)) {
+                    LOGGER.debug(
+                            "Skipping event for [{}] due to unsupported file extension of [{}].",
+                            filename,
+                            fileExtension);
+                    return;
+                }
+
+                if (changeListener != null) {
+                    waitForFileToBeCompletelyWritten(file);
+                    LOGGER.debug("Notifying [{}] of creation for file [{}].",
+                            changeListener.getClass()
+                                    .getName(),
+                            configurationDirectoryPath.resolve(filename));
+                    changeListener.notify(configurationDirectoryPath.resolve(filename));
+                }
+            } catch (InterruptedException e) {
+                LOGGER.debug("The [{}] was interrupted.",
```

Comment:
❗️ This is pretty bad if it happens because configuration files will not be processed anymore. We may want to bump up the logging level here (WARN?) and say that the application/bundle needs be restarted.